packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -221,4 +229,33 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
         }
     }
 
+    /**
+     * Detect the minimum indentation and type of indentation to be used
+     * as formatting options when using `jsoncparser`.
+     * @param text the preference text editor document.
+     *
+     * @returns a tuple representing the indentation size, and if it spaces.
+     */
+    protected detectIndentation(text: TextEditorDocument): [number, boolean] {
```

Comment:
we should not do it like that, but rather get it from editor options

Reply:
I implemented it this way because I could not find a way to get the information through editor options or other means.Any ideas?

Reply:
Why don't get them from the preference service? Even dirty editors should respect them and if they don't do then a user will commit some bogusly formatted code.

Reply:
True, in most cases the preferences values would be sufficient and only be inconsistent when the editor is dirty or a user chooses to update the values from the statusbar. I wanted to be smarter and try and determine the current indentation, but perhaps the preferences would be simpler.

Reply:
Using preference values should work for dirty editors, since editors always listening to the preference service and update options according. It can cause an issue if someone changes options from the status bar, but i would like to way till someone opens the real issue for it.

Reply:
> Using preference values should work for dirty editors, since editors always listening to the preference service and update options according. It can cause an issue if someone changes options from the status bar, but i would like to way till someone opens the real issue for it.

Sounds good, I'll make the updates.

Reply:
@akosyakov just a general question, is it possible to get a language specific preference?
In this case I only care about the corresponding JSON value for the preferences.

Reply:
`[json].my.preference`

Reply:
> `[json].my.preference`

Thanks!